i2c/ti/ads11xx.go

```diff
@@ -0,0 +1,221 @@
+// Package ti contains drivers for IC's produces by Texas Instruments.
+package ti
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/io/i2c"
+)
+
+type dataRate struct {
+	// sps is the data rate samples per second.
+	sps int
+
+	bitMask int
+
+	// size is the amount of bits that represent the output code
+	size uint
+}
+
+type ads11xx struct {
+	Conn *i2c.Device
+	Vref float64
+
+	dataRate dataRate
+	pga      int
+
+	// dataRates is a map that holds all valid values for data rate.
+	dataRates []dataRate
+}
+
+func newADS11xx(conn *i2c.Device, vref float64, dataRate, pga int, dataRates []dataRate) (ads11xx, error) {
+	a := ads11xx{
+		Conn:      conn,
+		Vref:      vref,
+		dataRates: dataRates,
+	}
+
+	if err := a.setDataRate(dataRate); err != nil {
+		return a, err
+	}
+
+	if err := a.SetPGA(pga); err != nil {
+		return a, err
+	}
+
+	return a, nil
+}
+
+// Voltage queries the channel of an ADC and returns its voltage.
+func (a ads11xx) Voltage(channel int) (float64, error) {
+	code, err := a.OutputCode(channel)
+	if err != nil {
+		return 0, err
+	}
+
+	max := math.Pow(2, float64(a.dataRate.size))
+	return ((a.Vref / max) * float64(code) / float64(a.pga)), nil
+}
+
+// OutputCode queries the channel and returns its digital output code. The
+// maximum code depends on the selected data rate. How higher the data rate,
+// how lower the amount of bits used.
```

Comment:
The comment seems very Dutch (hoe = how). In English, it should probably be `The higher the data rate, the lower the number of bits used.`